x/slashing/internal/keeper/infractions.go

```diff
@@ -31,20 +30,29 @@ func (k Keeper) HandleValidatorSignature(ctx sdk.Context, addr crypto.Address, p
 	index := signInfo.IndexOffset % k.SignedBlocksWindow(ctx)
 	signInfo.IndexOffset++
 
+	votearray := k.GetVoteArray(ctx, consAddr)
+	if votearray == nil {
+		votearray = types.NewVoteArray(k.SignedBlocksWindow(ctx))
+	} else if votearray.Size() != k.SignedBlocksWindow(ctx) {
```

Comment:
I believe the previous slashing code did correctly handle param-change-proposals for signed-blocks-window, so that may be what this is for. Agreed on documentation.

Reply:
yes, this is what i was thinking about